modules/core/src/main/java/org/apache/synapse/mediators/builtin/LogMediator.java

```diff
@@ -197,11 +197,14 @@ private String getFullLogMessage(MessageContext synCtx) {
         sb.append(getSimpleLogMessage(synCtx));
         try {
             org.apache.axis2.context.MessageContext a2mc = ((Axis2MessageContext) synCtx).getAxis2MessageContext();
-            if (JsonUtil.hasAJsonPayload(a2mc)) {
-                sb.append(separator).append("Payload: ").append(JsonUtil.jsonPayloadToString(a2mc));
-            } else if (synCtx.getEnvelope() != null) {
+            // If there is a XML element which store sourced JSON payload, synCtx.getEnvelope() will not be null.
+            // So need to check it also to avoid log in JSON format i.e. {"value" : "test"}
+            if (synCtx.getEnvelope() != null) {
                 sb.append(separator).append("Envelope: ").append(synCtx.getEnvelope());
```

Comment:
This fix is not correct. This will not print JSON, even it is a real native JSON payload as synCtx.getEnvelope() will never become null.